dirty-flag/src/main/java/com/iluwatar/dirtyflag/DataFetcher.java

```diff
@@ -0,0 +1,71 @@
+package com.iluwatar.dirtyflag;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A mock database manager -- Fetches data from a raw file.
+ * 
+ * @author swaisuan
+ *
+ */
+public class DataFetcher {
+
+  private static DataFetcher df;
+  private final String filename = "world.txt";
+  private long lastFetched = -1;
+
+  private DataFetcher() {
+  }
+
+  /**
+   * Init.
+   * 
+   * @return DataFetcher instance
+   */
+  public static DataFetcher getInstance() {
```

Comment:
Question: I don't understand the intent to keep `World` and `DataFetcher` singleton here. Any particular reason you chose this way?